### PR TITLE
add AltScores to ReportOptions in aggregate service

### PR DESCRIPTION
**Pull request by malaffoon**
This makes the changes to only the aggregate-service (with a little collateral damage). Nothing has been done in the webapp itself.

@gregcharles, this should result in the ReportOptions have the list of known alt-scores in the system.

### CODE REVIEW

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/AltScore.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.opentestsystem.rdw.olap.model;
```
Comment (malaffoon) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/AltScore.java:1]: For this purpose, AltScore and Claim are identical in content. I chose to make them separate things (with separate services, caches, etc.) to keep things a bit more specific.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/ScorableEntity.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package org.opentestsystem.rdw.olap.model;
```
Comment (malaffoon) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/model/ScorableEntity.java:1]: Factor out the common behavior of Claim and AltScore.

File: aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/ClaimRepository.java [hunk 3]
```diff
@@ -1,6 +1,6 @@
-package org.opentestsystem.rdw.reporting.common.repository;
```
Comment (malaffoon) [inline aggregate-service/src/main/java/org/opentestsystem/rdw/olap/repository/ClaimRepository.java:1]: I'm not quite sure why Claim and ClaimRepository were in common, they are only used in the aggregate-service. I moved them; we can always put them back if they do end up being used in more than one place.

File: aggregate-service/src/main/resources/application.sql.yml [hunk 4]
```diff
@@ -41,13 +48,12 @@ sql:
       SELECT id, code FROM asmt_type ORDER BY id
 
   claim:
-    aggregate:
```
Comment (malaffoon) [inline aggregate-service/src/main/resources/application.sql.yml:44]: this extra level was to differentiate from the non-aggregate implementation ... which wasn't used anywhere.

File: build.gradle [hunk 5]
```diff
@@ -123,7 +123,7 @@ subprojects {
         implementation 'org.slf4j:slf4j-api'
 
         // handy for creating metadata with @ConfigurationProperties
-        compileOnly 'org.springframework.boot:spring-boot-configuration-processor'
+        implementation 'org.springframework.boot:spring-boot-configuration-processor'
```
Comment (malaffoon) [inline build.gradle:126]: This shouldn't be necessary, but it helped get rid of some red in IntelliJ.

File: common/src/main/java/org/opentestsystem/rdw/reporting/common/util/CollectionUtils.java [hunk 6]
```diff
@@ -113,6 +113,14 @@
...
     }
 
+    /**
+     * @param value collection to check
+     * @return true if collection is null or empty
+     */
+    public static boolean safeIsEmpty(final Collection<?> value) {
+        return value == null || value.isEmpty();
+    }
+
```
Comment (gregcharles) [inline common/src/main/java/org/opentestsystem/rdw/reporting/common/util/CollectionUtils.java:123]: Why did you need this? It seems the same as isNullOrEmpty() ... except for declaring the parameter to be final, and Collection<?> in place of Collection<T>. So it's functionally the same, but different in documentation maybe? Could you just change isNullOrEmpty() to use a final parameter? That wouldn't break anything, or would it?

Comment (malaffoon) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/util/CollectionUtils.java:123]: I removed isNullOrEmpty because we don't use the amazon utilities anywhere else (and i really don't like using it since it is redundant with a few other, more common libraries). I was going to pull in apache commons collection4, but just did this instead. I could pull in that library if you'd rather.

Comment (gregcharles) [reply common/src/main/java/org/opentestsystem/rdw/reporting/common/util/CollectionUtils.java:123]: Ah, my mistake. I was looking at isNullOrEmpty() in CollectionUtils and missed that it wasn't _our_ CollectionUtils. I vaguely prefer using Apache since we do in some other places, but it's one more thing to import. I wish Java would just add this. (Of course by Java 14, maybe they have.)
